cmd/cue/cmd/testdata/script/cmd_errcode.txtar

```diff
@@ -16,4 +19,5 @@ command: errcode: {
 		kind:   "exec"
 		cmd:    "ls --badflags"
 		stderr: string // suppress error message
+		success: true
```

Comment:
Interesting, so any users relying on the current behavior where `cue cmd foo` fails if an `exec.Run` command returns a non-zero exit code would need to add a line like this.

That is technically a breaking change, and I wonder how many users it would affect in practice. Then again, the `success` field can't be of any use without a fix like this, like Roger pointed out in the original bug report, so I think we want to make the change anyway.

One thought outloud is that we could later also consider something like `exec.MustRun` which would be a shorthand for `exec.Run & {success: true}`.

Reply:
Ah yeah - that makes sense to me in order avoid the breaking change and also keep the same default.  Happy to add that in if there's alignment here 🙂 

One thing I am thinking about is assuming we would still want to have a fatal error when `cmd.Run` returns an error that isn't of type `os.ExitError` (as done in this PR currently), I'm thinking `mustSucceed: false` might be clearer than `allowFailure: true` since the latter may make it seem like that command would never fail?  

However, if we want to disregard all errors when a flag is set then maybe setting `allowFailures: true` would make more sense. Let me know what you all think / if there's a preference around that.

Reply:
Apologies if I'm missing something,  but may I suggest that the field not be a boolean, but an int instead - representing the command's exit code? 

That way, a default of "0" could be applied to keep the current backwards-compatible behaviour, whilst allowing the user control over which non-zero exit codes are permitted, via standard CUE number bounds, or a disjunction, or any other CUE-y mechanism.

Reply:
@nickfiggins please hold off on updating this PR - we had a brief internal discussion on what kind of API change we should make here, so I'll polish up those notes and share early next week :)

Reply:
> Apologies if I'm missing something, but may I suggest that the field not be a boolean, but an int instead - representing the command's exit code?

I believe the reason the `tool/exec` API does not expose the exit code is the same reason it doesn't appear in the https://pkg.go.dev/os/exec API: namely the behaviour around exit codes is OS-specific. @mvdan might well be able to offer more thoughts/colour on that point.

Reply:
Correct; exit codes are relatively common in Unix-like and POSIX-compliant systems, and Go does seem to support them on all platforms including Windows if you reach for the lower-level https://pkg.go.dev/os#ProcessState.ExitCode API, but they aren't a portable standard that is well understood and should replace `success: bool`.

That said, we could still expose the exit code alongside the success boolean, if that is useful to some users who for example run on Linux. If anyone has need for that, please raise an issue and we can discuss a change.